### PR TITLE
Add -one_shot_format to support Prometheus one shot output

### PR DESCRIPTION
**Pull request by roidelapluie**
This provides an easier to develop mtail programs because the Prometheus
exposition format is more concise.

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: cmd/mtail/main.go [hunk 1]
```diff
@@ -227,6 +235,22 @@ func main() {
...
+			}
+			err = e.Write(os.Stdout)
+			if err != nil {
+				glog.Error(err)
+				cancel()
+				os.Exit(1) //nolint:gocritic // false positive
+			}
+			cancel()
+			os.Exit(0) //nolint:gocritic // false positive
+		}
```
Comment (SuperQ) [inline cmd/mtail/main.go:253]: Should this have an `} else {` to not print the json output when using `prometheusOutput`?

File: cmd/mtail/main.go [hunk 2]
```diff
@@ -186,6 +191,9 @@ func main() {
 	if *oneShot {
 		opts = append(opts, mtail.OneShot)
 	}
+	if *prometheusOutput {
+		opts = append(opts, mtail.PrometheusOutput)
```
Comment (SuperQ) [inline cmd/mtail/main.go:195]: Maybe put this before `if *oneShot {` and set `oneShot` to true?

**Comment by roidelapluie** [comment]
changed to -format

File: cmd/mtail/main.go (hunk 1 above)
Comment (SuperQ) [reply cmd/mtail/main.go:260]: :+1:

**Review by SuperQ — APPROVED** [review]
LGTM

**Comment by jaqx0r** [comment]
I love the idea, and thanks @SuperQ for reviewing.  If I knew no-one depended on it, I'd make this the default and remove the JSON output (which I dislike.)

File: cmd/mtail/main.go [hunk 3]
```diff
@@ -46,11 +47,12 @@ var (
...
 
 	// Compiler behaviour flags.
-	oneShot      = flag.Bool("one_shot", false, "Compile the programs, then read the contents of the provided logs from start until EOF, print the values of the metrics store and exit. This is a debugging flag only, not for production use.")
-	compileOnly  = flag.Bool("compile_only", false, "Compile programs only, do not load the virtual machine.")
-	dumpAst      = flag.Bool("dump_ast", false, "Dump AST of programs after parse (to INFO log).")
-	dumpAstTypes = flag.Bool("dump_ast_types", false, "Dump AST of programs with type annotation after typecheck (to INFO log).")
-	dumpBytecode = flag.Bool("dump_bytecode", false, "Dump bytecode of programs (to INFO log).")
+	oneShot       = flag.Bool("one_shot", false, "Compile the programs, then read the contents of the provided logs from start until EOF, print the values of the metrics store in the given format and exit. This is a debugging flag only, not for production use.")
+	oneShotFormat = flag.String("format", "json", "Format to use with -one_shot. This is a debugging flag only, not for production use. Supported formats: json, prometheus.")
```
Comment (jaqx0r) [inline cmd/mtail/main.go:51]: Please make the flag `one_shot_format`.

**Review by jaqx0r — CHANGES_REQUESTED** [review]

File: cmd/mtail/main.go (hunk 3 above)
Comment (roidelapluie) [reply cmd/mtail/main.go]: Done !

**Comment by SuperQ** [comment]
We could always add the new option, and flip the default in a later release with some warning.

**Review by jaqx0r — APPROVED** [review]

**Pull request merged**